### PR TITLE
Add String Mapping Op

### PR DESCRIPTION
**Pull request by apaniukov**
**Ticket:** 132440

### CODE REVIEW

File: python/openvino_tokenizers/tokenizer_pipeline.py [hunk 1]
```diff
@@ -274,10 +274,6 @@ def byte_level_splitter(cls) -> "RegexSplitStep":
             behaviour="isolate",
         )
 
-    @classmethod
-    def add_whitespace_to_the_next_word(cls):
-        return cls(r"\s\S", invert=False, behaviour="merge_with_next")
-
```
Comment (rkazants) [inline python/openvino_tokenizers/tokenizer_pipeline.py:280]: do we need to remove it? please check

Comment (apaniukov) [reply python/openvino_tokenizers/tokenizer_pipeline.py:280]: Yes, but not necessary in this PR.

File: src/vocab_encoder.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+
+VocabEncoder::VocabEncoder (
+    const ov::OutputVector& arguments,
+    std::shared_ptr<std::map<std::vector<uint8_t>, int>> vocab,
+    int default_value
+) :
+    ov::op::Op(arguments), m_vocab(vocab), m_default_value(default_value) {
+    if (m_vocab == nullptr) {
+        auto packed_vocab_const = as_type_ptr<Constant>(arguments[3].get_node_shared_ptr()->get_input_node_shared_ptr(0));
```
Comment (rkazants) [inline src/vocab_encoder.cpp:25]: please fix this to use unpacked string tensor format for vocab

Comment (rkazants) [reply src/vocab_encoder.cpp:25]: Also, this input may be not necessarily  of Constant type

File: src/vocab_encoder.cpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    FRONT_END_GENERAL_CHECK(this->get_input_element_type(6) == element::i32, "Expected an i32 tensor for VocabEncode values.");
+    // vocab.size == vocab_values.size when vocab is static
+    FRONT_END_GENERAL_CHECK(
+        this->get_input_partial_shape(3).is_dynamic() || this->get_input_partial_shape(3) == this->get_input_partial_shape(6),
+        "Expected equal number of vocab keys and values."
+    );
+    // default value
+    FRONT_END_GENERAL_CHECK(this->get_input_element_type(7) == element::i32, "Expected an i32 scalar for VocabEncode default value.");
+    // one data output, reuse ragged dimensions from split
+    this->set_output_type(0, element::i32, get_input_partial_shape(0));
```
Comment (rkazants) [inline src/vocab_encoder.cpp:65]: Please support `i64` type for values

**Review by rkazants — CHANGES_REQUESTED** [review]
Now it is not suitable for TensorFlow needs:
1. format of vocal keys needs to be changed
2. need support of i64 values

**Review by rkazants — APPROVED** [review]

**Pull request merged**